link-grammar/linkage/score.c

```diff
@@ -68,10 +69,54 @@ static float compute_disjunct_cost(Linkage lkg)
 	return lcost;
 }
 
+/// Compute the extra cost of a multi-connector being used multiple
+/// times. Any cost on that multi-connector needs to be added for each
+/// usage of it.
+static float compute_multi_cost(Linkage lkg)
+{
+	size_t lword = 0;
+	float mcost = 0.0;
+	for (size_t i = 0; i < lkg->num_words; i++)
+	{
+		Disjunct * dj = lkg->chosen_disjuncts[i];
+		if (NULL == dj) continue;
+
+		// Look at the right-going connectors.
+		for (Connector* rcon = dj->right; rcon; rcon=rcon->next)
+		{
+			// No-op if not a multi-connector.
+			if (false == rcon->multi) continue;
+
+			// Find the links using this connector.
+			for (size_t l = 0; l < lkg->num_links; l++)
+			{
+				// The link is not even for this word. Ignore it.
+				if (lword != lkg->link_array[l].lw) continue;
+
+				// Find the links that are using our multi-connector.
+				for (Connector* wrc = lkg->link_array[l].rc; wrc; wrc=wrc->next)
+				{
+					// Skip if no match.
+					if (strcmp(rcon->desc->string, wrc->desc->string)) continue;
```

Comment:
As I wrote in the general response (in the wrong box...), there is 1-1 match between condesct_t elements and connector strings, disregarding if the connectors are multi or not. Thus the condesc_t strings don't include`@`.
So there is a need for a different comparison.